crates/macros/src/implement_macro.rs

```diff
@@ -0,0 +1,167 @@
+                }
+            }
+            UseTree::Glob(input) => {
+                return Err(Error::new_spanned(input, "Glob syntax not supported"));
+            }
+            UseTree::Group(input) => {
+                for tree in &input.items {
+                    self.walk_implement(reader, tree, namespace)?;
+                }
+            }
+            UseTree::Rename(input) => {
+                return Err(Error::new_spanned(input, "Rename syntax not supported"));
+            }
+        }
+
+        Ok(())
+    }
+
+    fn parse_override(&mut self, reader: &'static TypeReader, cursor: ParseStream) -> Result<()> {
+        // Any number of methods may be overridden but only if a class is being overriden.
+        if let Some((namespace, name)) = self.extend {
+            if cursor.parse::<Token![override]>().is_ok() {
+                let methods = reader
+                    .resolve_type_def(namespace, name)
+                    .overridable_methods();
+
+                while let Ok(input) = cursor.parse::<Ident>() {
+                    if let Some(name) = methods.get(input.to_string().as_str()) {
+                        self.overrides.insert(name);
+                    } else {
+                        return Err(Error::new_spanned(
+                            input,
+                            "No overridable method with this name",
+                        ));
+                    }
+                }
+
+                if !cursor.is_empty() {
+                    cursor.parse::<Token![,]>()?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn parse_extend(&mut self, reader: &'static TypeReader, cursor: ParseStream) -> Result<()> {
+        // Only one class may be extended
+        if self.extend.is_none() && cursor.parse::<extend>().is_ok() {
+            self.walk_extend(reader, &cursor.parse()?, &mut String::new())?;
+
+            if !cursor.is_empty() {
+                cursor.parse::<Token![,]>()?;
+            }
+        }
+
+        Ok(())
+    }
+
+    fn walk_extend(
+        &mut self,
+        reader: &'static TypeReader,
+        tree: &UseTree,
+        namespace: &mut String,
+    ) -> Result<()> {
+        match tree {
+            UseTree::Path(input) => {
+                if !namespace.is_empty() {
+                    namespace.push('.');
+                }
+
+                namespace.push_str(&input.ident.to_string());
+                self.walk_extend(reader, &*input.tree, namespace)?;
+            }
+            UseTree::Name(input) => {
+                let name = input.ident.to_string();
+
+                if let Some((namespace, name)) = reader.get_type_name(namespace, &name) {
+                    if reader
+                        .resolve_type_def(namespace, name)
+                        .is_public_composable()
+                    {
+                        self.extend = Some((namespace, name));
```

Comment:
You could possibly use `fn replace()` here and assert that the result `.is_none()` to enforce the constraint above.